src/java/com/sarality/app/view/dialog/CustomViewDialogFragment.java

```diff
@@ -85,6 +85,8 @@ protected void reinitialize() {
    * @param loaderId Id to used for the loader
    */
   protected <D> void startLoad(DataSource<D> dataSource, int loaderId, DataConsumer<D> consumer) {
+    // Kill all previously loaded data by the activity
+    getActivity().getSupportLoaderManager().destroyLoader(loaderId);
```

Comment:
Activity -> Fragment -> loads it data with different loader Ids
Fragment -> Action -> DialogFragment -> now tries to load data with its loaderIds

The problem is that the dialogfragment does not know about what loaderIds were used by the fragment. It will have its own values ( in this case it was 0) which could clash.

I am open to ideas but having stale data around is not a good thing to have either. Moreover, the dataType in the result was different which actually caused a crash in my case.